src/pybind/mgr/dashboard/tools.py

```diff
@@ -21,6 +21,11 @@
 except ImportError:
     from urllib.parse import urljoin
 
+try:
+    from typing import Dict, Any  # pylint: disable=unused-import
```

Comment:
We're taking seriously this mypy thing, right? Interesting!

Reply:
> We're taking seriously this mypy thing, right? Interesting!

It already prevented a bug: https://github.com/ceph/ceph/pull/26547/files#r264247893 .
I don't know, if we want to go down that road. But, if we do want to, something like #26769 is unfortunately required :disappointed:

Reply:
Yup, I saw your PR, @sebastian-philipp, so kudos to you as well on this move. I'll support any type checking that can be discretionary (though exceptionally) disabled.

Reply:
yep, as of right now, it's purely optional

Reply:
Even if we had a type checker integration in make check, it still **could** be considered optional, as no one is enforced to write type hints, even though they are there.

Reply:
Well, the purpose of `make check` is to check and verify stuff, especially as make check doesn't print anything, if the test passes. 

So, how could an **optional** `make check` integration look like?

Reply:
The check itself wouldn't be optional, but it would be up to the developer to write new type hints or not. That's "optional" in a wide sense, though.

`make check` would then run `mypy` to verify the types and exit properly if everything is fine, otherwise return a non-zero exit code, which could be used to determine the failure of the test.

This would not enforce anyone to write type hints but it would enforce that the types, which are already there, are valid.

This is not a proposal, I just wanted to add that this would also be possible, assuming that the interpretation of "optional" leaves enough room for this implementation.